org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceExecuteCommandHandlerTest.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal.handlers;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import java.util.Arrays;
+
+import org.eclipse.jdt.ls.core.internal.managers.AbstractProjectsManagerBasedTest;
+import org.eclipse.lsp4j.ExecuteCommandParams;
+import org.junit.Test;
+
+public class WorkspaceExecuteCommandHandlerTest extends AbstractProjectsManagerBasedTest {
+
+	@Test
+	public void testExecuteCommand() {
+		WorkspaceExecuteCommandHandler handler = new WorkspaceExecuteCommandHandler();
+		ExecuteCommandParams params = new ExecuteCommandParams();
+		params.setCommand("testcommand1");
+		params.setArguments(Arrays.asList("hello", "world"));
+		Object result = handler.executeCommand(params, monitor);
+		assertEquals("testcommand1: helloworld0", result);
+
+		params.setCommand("testcommand2");
+		result = handler.executeCommand(params, monitor);
+		assertEquals("testcommand2: helloworld1", result);
+	}
+
+	@Test
+	public void testExecuteCommandNonexistingCommand() {
+		WorkspaceExecuteCommandHandler handler = new WorkspaceExecuteCommandHandler();
+		ExecuteCommandParams params = new ExecuteCommandParams();
+		params.setCommand("testcommand.not.existing");
+		params.setArguments(Arrays.asList("hello", "world"));
+		Object result = handler.executeCommand(params, monitor);
+		assertNull(result);
+	}
+
+	@Test
+	public void testExecuteCommandThrowsExceptionCommand() {
+		WorkspaceExecuteCommandHandler handler = new WorkspaceExecuteCommandHandler();
+		ExecuteCommandParams params = new ExecuteCommandParams();
+		params.setCommand("testcommand.throwexception");
+		Object result = handler.executeCommand(params, monitor);

```

Comment:
can we capture logging output?

Reply:
Yes, it turns out we can add the loggin listener to capture the exception information